cli/backend_base.py

```diff
@@ -212,13 +213,31 @@ def _ship_certs(self, saltmaster_ip):
 
         return platform_certs_tarball
 
+    def _check_volume_count(self, data_volume_count, volumes):
+        data_volumes = []
+        for volume in volumes:
+            if '/data' in volume:
+                data_volumes.append(volume)
+        if len(data_volumes) != data_volume_count:
+            CONSOLE.error('Data volumes are not equal to data volume count')
+            sys.exit(1)
+
     def _get_volume_info(self, node_type, config_file):
         volumes = None
         if node_type:
-            with open(config_file, 'r') as infile:
-                volume_config = yaml.load(infile)
-                volume_class = volume_config['instances'][node_type]
-                volumes = volume_config['classes'][volume_class]
+            config_file_list = config_file.split("/")
+            config_file_path = "%s/%s" % (config_file_list[0], config_file_list[1])
+            data_volume_count = int(self._pnda_env['datanode']['DATA_VOLUME_COUNT'])
+            device_root = self._pnda_env['datanode']['DEVICE_ROOT']
+            template_loader = jinja2.FileSystemLoader(searchpath=config_file_path)
+            template_env = jinja2.Environment(loader=template_loader)
+            template = template_env.get_template(config_file_list[2])
+            volume_config = yaml.safe_load(template.render(DATA_VOLUME_COUNT=data_volume_count, DEVICE_ROOT=device_root))
+            volume_class = volume_config['instances'][node_type]
+            volumes = volume_config['classes'][volume_class]
+            if 'hadoop-dn' in node_type:
```

Comment:
I don't think we need a check that the number of volumes matches the value in the yaml since the value in the yaml is used to determine the loop count in the first place.